lib/compress/zstd_compress_internal.h

```diff
@@ -828,6 +836,24 @@ size_t ZSTD_hashPtr(const void* p, U32 hBits, U32 mls)
     }
 }
 
+MEM_STATIC FORCE_INLINE_ATTR
+size_t ZSTD_hashPtrSalted(const void* p, U32 hBits, U32 mls, const U64 hashSalt) {
+    /* Although some of these hashes do support hBits up to 64, some do not.
+     * To be on the safe side, always avoid hBits > 32. */
+    assert(hBits <= 32);
+
+    switch(mls)
+    {
+        default:
+        case 4: return ZSTD_hash4PtrS(p, hBits, (U32)hashSalt);
```

Comment:
I would rather recommend to use the upper part of `hashSalt` (`(U32)(hashSalt >> 32)`),
as it's less predictable than the lower part.

An alternative approach could be to make `hashSalt` evolution of lower bits less predictable.